cmd/sonobuoy/app/worker.go

```diff
@@ -87,7 +90,22 @@ func loadAndValidateConfig() (*plugin.WorkerConfig, error) {
 	return cfg, nil
 }
 
+// sigHandler is used to manage graceful cleanups when a TERM signal is received.
+func sigHandler() <-chan struct{} {
+	stop := make(chan struct{})
+	go func() {
+		sigc := make(chan os.Signal, 1)
+		signal.Notify(sigc, syscall.SIGTERM)
+		<-sigc
+		// The worker needs to not shutdown immediately upon receiving SIGTERM.
+		time.Sleep(30)
```

Comment:
During a graceful shutdown k8s will send a TERM to all the containers of the pod. The pod contains 1) the worker 2) the e2e test pod. We sleep for a little bit before actually exiting to capture and transmit results.

I was looking into IPC but I wasn't sure it was worth the effort vs a slight delay in shutting down.

Reply:
Maybe make this 20 b/c termination timeout is 30.

Reply:
Shouldn't the worker hold for less then 30?  e.g. 20 in order to send the data back.

Reply:
Yeah, 20 probably makes more sense because of the broken case:

If we use 30 seconds: In a normal case I'd expect the plugin to write data to the done file and have the worker pick it up and send it out and clean itself up like normal (because it finished all the work it cared about). In the broken case where the results are not written out in 30 seconds, it's going to simultaneously send a message on the `stop` channel as well as receive a SIGKILL from k8s (because the grace period has expired). Sometimes we'd exit from a SIGKILL sometimes we'd exit cleanly from a message on the `stop` channel.

In the broken case, if 20 seconds elapsed and we still hadn't received results, we still have 10 seconds of graceful cleanup. We'd send a message on the stop channel signaling which would shut the server down and exit with a nil error status. If we still hadn't exited in 10 seconds (unlikely given the code path, but 💻s) then we'd get a SIGKILL and exit.